app/components/ITunesCard/index.js

```diff
@@ -0,0 +1,134 @@
+/**
+ *
+ * ITunesCard
+ *
+ */
+
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import truncate from 'lodash/truncate';
+import { isEmpty } from 'lodash';
+import { Card, Popover, Button, Progress } from 'antd';
+import { PlayCircleOutlined, PauseCircleOutlined } from '@ant-design/icons';
+import T from '@components/T';
+import If from '@components/If';
+import { colors } from '@app/themes';
+import { ACTIONS } from '@utils/constants';
+
+const CustomCard = styled(Card)`
+  && {
+    margin: 1.25em 0;
+    box-shadow: ${colors.boxShadow};
+  }
+`;
+const Wrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  margin-bottom: 0.75rem;
+  gap: 1em;
+`;
+const CustomImg = styled.img`
+  display: block;
+  margin: 0.75rem auto;
+  height: 10rem;
+`;
+
+export function ITunesCard({ track, onActionClick }) {
+  const audioRef = useRef(null);
+  const [progress, setProgress] = useState(0);
+  const [intervalStore, setIntervalStore] = useState(null);
+
+  useEffect(() => {
+    return () => {
+      clearInterval(intervalStore);
+    };
+  }, []);
+  useEffect(() => {
+    clearInterval(intervalStore);
+  }, [track]);
+
+  function calculateProgress() {
+    const intervalValue = setInterval(() => {
+      const fraction = (audioRef.current.currentTime / audioRef.current.duration) * 100;
+      setProgress(fraction);
+    }, 1000);
+    setIntervalStore(intervalValue);
+  }
+  const handleTrackPlay = (action) => {
+    if (action === ACTIONS.PLAY) {
+      calculateProgress();
+    } else {
+      clearInterval(intervalStore);
+    }
+    onActionClick(audioRef, action);
+  };
+
+  const text = <T id="details" type="subheading" marginBottom={5} />;
+  const content = (
+    <>
+      <If condition={track?.trackPrice} otherwise={<T data-testid="price-unavailable" id="track_price_unavailable" />}>
+        <T id="track-price" values={{ price: track?.trackPrice }} />
+      </If>
+      <If
+        condition={!isEmpty(track?.primaryGenreName)}
+        otherwise={<T data-testid="genre-unavailable" id="track_genre_unavailable" />}
+      >
+        <T id="genre-name" values={{ name: track?.primaryGenreName }} />
+      </If>
+    </>
+  );
+  return (
+    <CustomCard data-testid="itunes-card">
```

Comment:
please test whether all the items that we expect to be rendered are getting rendered

Reply:
okay